heat_pump_adoption_modelling/analysis/MCS/plot_mcs.py

```diff
@@ -0,0 +1,549 @@
+    ax.grid(color="0.8")
+
+    # Put a legend to the right of the current axis
+    box = ax.get_position()
+    ax.set_position([box.x0, box.y0, box.width * 0.8, box.height])
+    ax.legend(
+        title="Flow temp. ($\degree$C)", loc="center left", bbox_to_anchor=(1, 0.5)
+    )
+
+    plt.savefig("outputs/figures/mean_scop.png")
+
+
+def ashp_capacity_cost_boxplot(df):
+    """Produce boxplot of ASHP inflation-adjusted installation costs
+    vs their capacities for capacities between 5 and 16 inclusive
+    (the most common capacities), only considering data from 2019 onwards
+    (the most reliable cost data).
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe containing "year", "tech_type", "capacity" and
+        "inflated_cost" variables.
+
+    Return
+    ----------
+        No objects returned, but figure is saved as capacity_cost_boxplots.png.
+    """
+
+    # Filter to only recent and ASHP data
+    plottable_data = df.loc[
+        (df["year"] >= 2019) & (df["tech_type"] == "Air Source Heat Pump")
+    ]
+
+    capacity_list = range(5, 17)
+
+    # Form a list of lists of cost data for each capacity
+    data_list = []
+    for capacity in capacity_list:
+        filtered_df = plottable_data.loc[plottable_data["capacity"] == capacity]
+        costs = list(filtered_df["inflated_cost"].dropna())
+        data_list.append(costs)
+
+    fig, ax = plt.subplots()
+
+    ax.boxplot(
+        data_list,
+        flierprops={
+            "marker": "o",
+            "markersize": 3,
+            "markerfacecolor": "black",
+            "alpha": 0.3,
+        },
+    )
+
+    ax.grid(axis="y", color="0.8")
+    ax.set_ylim(0, 50000)
+    ax.set_xlabel("HP capacity")
+    ax.set_ylabel("Inflation-adjusted cost (£2021)")
+    ax.set_title("MCS-certified ASHP installation costs by capacity (2019-21)")
+
+    # Add labels to x axis
+    plt.setp(ax, xticklabels=capacity_list)
+    plt.tight_layout()
+
+    plt.savefig(PROJECT_DIR / "outputs/figures/capacity_cost_boxplots.png")
+
+
+def ashp_archetypes_boxplot(df, variable):
+    """Produce boxplot of ASHP inflation-adjusted installation costs
+    for each archetype (only considering data from 2019 onwards).
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe containing "year", "tech_type",
+        "PROPERTY_TYPE", "BUILT_FORM" and "built_post_1950" variables,
+        as well as the variable specified in the "variable" parameter.
+    variable : str
+        Tested with either "inflated_cost" or "capacity".
+
+    Return
+    ----------
+        No objects returned, but figure is saved as ashp_cost_archetype_boxplots.png
+        if variable is "inflated_cost" and ashp_capacity_archetype_boxplots.png if
+        variable is "capacity".
+    """
+
+    ashp_data_list = []
+
+    archetype_dict = generate_archetype_dict(df)
+    # off_scale = 0
+    # total = 0
+
+    for info in archetype_dict.values():
```

Comment:
The ordering of the different archetypes seems a bit random.

![Screenshot 2021-12-03 at 16 08 11](https://user-images.githubusercontent.com/42718928/144636059-0b2da9e8-d4a9-4162-8062-cbdf41537bbc.png)

Instead of interating over the values of a dict (which is a bit unpredictable), better use a tuple list as described above. This would translate to something like:

```suggestion
for name, condition in enumerate(archetypes):
        filtered_df = df.loc[
            (df["year"] >= 2019)
            & (df["tech_type"] == "Air Source Heat Pump")
            &  condition
  
```

Reply:
This is actually the intended ordering of the archetypes (increasing expected HP capacity) but completely agree with the idea of using a list of tuples instead!

Reply:
Oh, I see, my bad! ;-) Now I get why you named them archetype_1 etc...